### PR TITLE
Cope with '%' in password when waiting for migrations

### PR DESCRIPTION
**Pull request by highfly22**
---

When there is a '%' in the sql connection url, wait-for-airflow-migrations container complains no permission to connect database.
This commit fix that issue by replacing '%' with '%%'.

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: chart/templates/_helpers.yaml [hunk 1]
```diff
@@ -380,7 +380,7 @@ server_tls_key_file = /etc/pgbouncer/server.key
         directory = os.path.join(package_dir, 'migrations')
         config = Config(os.path.join(package_dir, 'alembic.ini'))
         config.set_main_option('script_location', directory)
-        config.set_main_option('sqlalchemy.url', settings.SQL_ALCHEMY_CONN)
+        config.set_main_option('sqlalchemy.url', settings.SQL_ALCHEMY_CONN.replace('%', '%%'))
```
Comment (potiuk) [inline chart/templates/_helpers.yaml:383]: Isn't that a problem of the wrong URL specified?

For the SQL alchemy connection should be URL-encoded in the first place. This is true for the individual parameters of the query but also for user and password. So I think your problem is that in your original URL you should us %25 where you have '%'  and not replace % with %%.

**Review by potiuk — CHANGES_REQUESTED** [review]
I thin the proposed solutin is not good  - see the comment.

File: chart/templates/_helpers.yaml (hunk 1 above)
Comment (ashb) [reply chart/templates/_helpers.yaml:383]: The issue is I think that '%' in ConfigParser is special -- it's the same case as I fixed a few years ago in https://github.com/apache/airflow/pull/3787.

**Review by ashb — APPROVED** [review]

**Comment by potiuk** [comment]
I see!

**Review by potiuk — APPROVED** [review]

**Pull request merged**